packages/data-sdk/src/Device.ts

```diff
@@ -274,7 +274,13 @@ export class Device extends EventEmitter implements IRealtimeDevice {
 
       // We can connect our real-time communication client to device peers by their ID
       this.remoteDevicePeerId = devicePeer.id;
-      await (rtcClient as RtcClient).connect(this.remoteDevicePeerId);
+      let isConnected = null;
+      while (typeof isConnected !== "string") {
+        isConnected = await (rtcClient as RtcClient).connect(
+          this.remoteDevicePeerId
```

Comment:
Seems like this would cause an infinite loop if the device ever goes offline and its peer ID expires. Since `this.remoteDevicePeerId` is updated on line 276, it will never be refreshed once in this loop.

Maybe every X failed attempts, it would make sense to execute this logic. Could just break it out as a separate method. https://github.com/FormantIO/toolkit/pull/78/files#diff-f777f3fac8c979f45751b42876eea470219441adcb722899e6022fd7ca85cf73R265-R276

Reply:
it does cause an infinite loop, I'll update it to try a few times. and just for some context, this issue happens when we try to start a `observe` session, when a teleop session is connected already, it fails like 50% of the time.